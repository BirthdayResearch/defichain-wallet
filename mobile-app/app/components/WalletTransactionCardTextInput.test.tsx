import { render } from "@testing-library/react-native";
import {
  InputType,
  WalletTransactionCardTextInput,
} from "./WalletTransactionCardTextInput";

const WalletInputType: InputType[] = ["default", "numeric"];

describe("wallet transactional card text input", () => {
  WalletInputType.forEach((type) => {
    it(`should render with ${type} keyboard`, async () => {
      const value = "";
      const onClear = jest.fn;
      const rendered = render(
        <WalletTransactionCardTextInput
          title="foo"
          titleTestID="titleTestID"
          placeholder="bar"
          testID="testID"
          value={value}
          valid={false}
          inlineText={{
            type: "error",
            text: "invalid",
          }}
          inputType={type}
          displayClearButton={value !== ""}
          onClearButtonPress={onClear}
        />
      );
      expect(rendered.toJSON()).toMatchSnapshot();
    });
  });
});
