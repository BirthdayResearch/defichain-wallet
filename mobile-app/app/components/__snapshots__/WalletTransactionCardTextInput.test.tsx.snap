// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wallet transactional card text input should render with default keyboard 1`] = `
<View
  style={
    [
      {},
      {},
    ]
  }
>
  <Text
    style={
      [
        {},
        [
          {},
          {},
        ],
      ]
    }
    testID="titleTestID"
    text="foo"
  >
    foo
  </Text>
  <View
    style={
      [
        {},
        {},
      ]
    }
  >
    <View
      style={
        [
          {},
          false,
          undefined,
        ]
      }
    >
      <TextInput
        editable={true}
        keyboardType="default"
        onBlur={[Function]}
        placeholder="bar"
        placeholderTextColor="#ffffff"
        selectionColor="#ffffff"
        style={
          [
            {},
            {},
          ]
        }
        testID="testID"
        value=""
      />
    </View>
    <View />
  </View>
  <Text
    style={
      [
        {},
        undefined,
      ]
    }
    testID="testID_error"
  >
    invalid
  </Text>
</View>
`;

exports[`wallet transactional card text input should render with numeric keyboard 1`] = `
<View
  style={
    [
      {},
      {},
    ]
  }
>
  <Text
    style={
      [
        {},
        [
          {},
          {},
        ],
      ]
    }
    testID="titleTestID"
    text="foo"
  >
    foo
  </Text>
  <View
    style={
      [
        {},
        {},
      ]
    }
  >
    <View
      style={
        [
          {},
          false,
          undefined,
        ]
      }
    >
      <TextInput
        editable={true}
        keyboardType="default"
        onBlur={[Function]}
        placeholder="bar"
        placeholderTextColor="#ffffff"
        selectionColor="#ffffff"
        style={
          [
            {},
            {},
          ]
        }
        testID="testID"
        value=""
      />
    </View>
    <View />
  </View>
  <Text
    style={
      [
        {},
        undefined,
      ]
    }
    testID="testID_error"
  >
    invalid
  </Text>
</View>
`;
